runtime/compiler/control/J9CompilationStrategy.cpp

```diff
@@ -252,7 +253,18 @@ J9::CompilationStrategy::processInterpreterSample(TR_MethodEvent *event)
       fe->printTruncatedSignature(sig, SIG_SZ, (TR_OpaqueMethodBlock*)j9method);
 
       if (logSampling)
-         curMsg += sprintf(curMsg, "(%d)\tInterpreted %s\t", totalSampleCount, sig);
+         {
+         size_t msgLen = snprintf(curMsg, maxMsgLen, "(%d)\tInterpreted %s\t", totalSampleCount, sig);
+         if (msgLen <= maxMsgLen)
```

Comment:
Many occurrences of this pattern need to be updated to properly handle `snprintf()` returning a negative value:
```c
         int msgLen = snprintf(curMsg, maxMsgLen, "(%d)\tInterpreted %s\t", totalSampleCount, sig);
         if ((0 < msgLen) && ((size_t)msgLen <= maxMsgLen))
```